src/setting.js

```diff
@@ -3,8 +3,15 @@ const fs = require('fs')
 const path = require('path')
 const {app, BrowserWindow} = require('electron')
 
+const portableDir = process.env.PORTABLE_EXECUTABLE_DIR
+if (portableDir) {
+  confirmPath = portableDir
+} else {
+  confirmPath = app.getPath('userData')
+}
+
 for (let dir of [
-  (exports.userDataDirectory = app.getPath('userData')),
+  (exports.userDataDirectory = confirmPath),
```

Comment:
The variable `confirmPath` is not defined. We could also inline the logic here:

```suggestion
  (exports.userDataDirectory = process.env.PORTABLE_EXECUTABLE_DIR || app.getPath('userData')),
```